src/process.rs

```diff
@@ -0,0 +1,130 @@
+/// Select the best match in a list or dictionary of choices.
+///
+/// Find best matches in a list or dictionary of choices, return a generator of tuples containing
+/// the match and its score. If a dictionary is used, also returns the key for each match.
+///
+/// TODO: Add support for choices as HashMap<&str, &str>, not only as slice &[&str].
+pub fn extract_without_order<'a, I, T>(
+    query: &str,
+    choices: I,
+    processor: &dyn Fn(&str, bool) -> String,
+    scorer: &dyn Fn(&str, &str, bool, bool) -> u8,
+    score_cutoff: u8,
+) -> Vec<(String, u8)>
+where
+    I: IntoIterator<Item = T>,
+    T: AsRef<str>,
+{
+    let processed_query: String = processor(query, false);
+    if processed_query.is_empty() {
+        println!("Applied processor reduces input query to empty string, all comparisons will have score 0. [Query: '{0}']", processed_query.as_str());
```

Comment:
I think it's a useful warning to have, so I'll comment this out and mark it as // TODO: to make sure warnings and/or verbose loggings are available options to be enabled when using the library (this can be done when the input parameters to `process::*` are moved to option structs to provide default parameters for `processor`, `scorer`, and `score_cutoff`).